astropy/utils/tests/test_misc.py

```diff
@@ -86,6 +86,27 @@ def test_JsonCustomEncoder():
     assert newd == tmpd
 
 
+def test_JsonCustomEncoder_FITS_rec_VLF():
+    from astropy.io import fits
+
+    col1 = fits.Column(
+        name='var', format='PI()',
+        array=np.array([[1,2,3], [11, 12]], dtype=np.object_))
+
+    col2 = fits.Column(
+        name='i2', format='2I',
+        array=np.array([[101,102], [111, 112]], dtype=np.object_))
+
+    hdu = fits.BinTableHDU.from_columns([col1, col2])
+    data = hdu.data
+
+    #what is converted by np.recarray.tolist is in fact descr_output, [3, 0], [2, 6] - n items and offset
+    assert json.dumps(np.recarray.tolist(data), cls=misc.JsonCustomEncoder) == '[[[3, 0], [101, 102]], [[2, 6], [111, 112]]]'
+    assert json.dumps(data['var'], cls=misc.JsonCustomEncoder) == '[[1, 2, 3], [11, 12]]'
+    assert json.dumps(data['i2'], cls=misc.JsonCustomEncoder) == '[[101, 102], [111, 112]]'
+    assert json.dumps(data, cls=misc.JsonCustomEncoder) == '[[[1, 2, 3], [101, 102]], [[11, 12], [111, 112]]]'
+
```

Comment:
Could you test first that `data.tolist()` is returning the expected result ? And move the test to `io.fits`.

Reply:
Also the `i2` column should not have an object dtype.

Reply:
> Also the `i2` column should not have an object dtype.

good point, thanks.

Reply:
> Could you test first that `data.tolist()` is returning the expected result ? And move the test to `io.fits`.

It looks like tolist() is not explicitly tested or even particularly used in `io.fits` yet. I am adding a test in there. There is an additional complexity which I did not intend to address at this time, since it is has no impact  on json, but I will address it now.

So I wanted to have this json serialization comparison test which is independent of particular implementation. 
Should I keep this json comparison? If yes, should I really move it to `io.fits`?  Since jsonification is tested in utils.

Reply:
`tolist` is part of the API of `.data` tables, so making sure it works also with VLA is a good thing to do.

Re- json test, yes I think it's useful. Both make sense actually, testing that fits objects can be serialized to json or that the `JsonCustomEncoder` can serialize fits objects. But currently there is no test with fits in `test_misc.py` and the VLA is a quite specific feature. It would actually be useful to have a more general test for `JsonCustomEncoder` and fits, with different types of columns. Either location is fine to me.

Reply:
Ok, I started to make a test with some column variants [here](https://github.com/oda-hub/astropy/blob/ef1b4bf71/astropy/utils/tests/test_misc.py#L89-L113). Please let me know if it does not fit your intent, otherwise I will continue later.